docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,491 @@
+
+If a backend library supports multiple domains (i.e. BLAS, LAPACK, DFT, RNG), it may be desirable to only enable selected domains.
+For this, the ``TARGET_DOMAINS`` variable should be set.
+See the section `TARGET_DOMAINS`_.
+
+By default, the library also additionally builds examples and tests.
+These can be disabled by setting the parameters ``BUILD_FUNCTIONAL_TESTS`` and ``BUILD_EXAMPLES`` to off.
+Building the functional tests may require additional external libraries.
+See the section `Building and running tests`_ for more information.
+
+The most important supported build options are:
+
+.. list-table::
+   :header-rows: 1
+
+   * - CMake Option
+     - Supported Values
+     - Default Value 
+   * - ENABLE_MKLCPU_BACKEND
+     - True, False
+     - True      
+   * - ENABLE_MKLGPU_BACKEND
+     - True, False
+     - True      
+   * - ENABLE_CUBLAS_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_CUSOLVER_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_CUFFT_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_CURAND_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_NETLIB_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_ROCBLAS_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_ROCFFT_BACKEND
+     - True, False
+     - False    
+   * - ENABLE_ROCSOLVER_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_ROCRAND_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_MKLCPU_THREAD_TBB
+     - True, False
+     - True      
+   * - ENABLE_PORTBLAS_BACKEND
+     - True, False
+     - False      
+   * - ENABLE_PORTFFT_BACKEND
+     - True, False
+     - False      
+   * - BUILD_FUNCTIONAL_TESTS
+     - True, False
+     - True      
+   * - BUILD_EXAMPLES
+     - True, False
+     - True      
+   * - TARGET_DOMAINS (list)
+     - blas, lapack, rng, dft, sparse_blas
+     - All domains 
+
+Some additional build options are given in the section `Additional build options`_.
+
+.. _build_target_domains:
+
+TARGET_DOMAINS
+^^^^^^^^^^^^^^
+
+OneMKL supports multiple domains: BLAS, DFT, LAPACK, RNG and sparse BLAS.
+The domains built by oneMKL can be selected using the ``TARGET_DOMAINS`` parameter.
+In most cases, ``TARGET_DOMAINS`` is set automatically according to the domains supported
+by the backend libraries enabled.
+However, while most backend libraries support only one of these domains, but some may support multiple.
+For example, the ``MKLCPU`` backend supports every domain. To enable support for only the BLAS domain in
+the oneMKL interface library whilst compiling with ``MKLCPU``, ``TARGET_DOMAINS`` could be set to ``blas``. 
+To enable BLAS and DFT, ``-DTARGET_DOMAINS="blas dft"`` would be used.
+
+
+.. _build_for_intel_onemkl_dpcpp:
+
+Building for Intel(R) oneMKL
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The closed-source Intel(R) oneMKL supports multiple domains on both x86 CPUs and Intel GPUs.
```

Comment:
```suggestion
The Intel(R) oneMKL backend supports multiple domains on both x86 CPUs and Intel GPUs.
```

Reply:
Done in [931ee0f](https://github.com/oneapi-src/oneMKL/pull/510/commits/931ee0f68ff1bd61a5f9f7c1957559a4a41a2215).